include/swift/SILOptimizer/Utils/Differentiation/PullbackEmitter.h

```diff
@@ -354,6 +361,13 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
   ///                     field in tangent space corresponding to #field
   void visitStructExtractInst(StructExtractInst *sei);
 
+  /// Handle `ref_element_addr` instruction.
+  ///   Original: y = ref_element_addr x, <n>
+  ///    Adjoint: adj[x] += struct (0, ..., #field': adj[y], ..., 0)
+  ///                                       ^~~~~~~
+  ///                     field in tangent space corresponding to #field
+  void visitRefElementAddrInst(RefElementAddrInst *reai);
```

Comment:
We should get rid of this header soon...

Reply:
I can do it soon :)